homemade_functions.py

```diff
@@ -0,0 +1,102 @@
+import random
+from random import randint
+from constants import*
+
+
+def convert_to_list(word):
+    deconstructed_word = []
+    for letter in word:
+        deconstructed_word.append(letter.upper())
+    return deconstructed_word
+
+
+def list_length(list_to_count):
+    count = 0
+    for i in list_to_count:
+        count += 1
+    return count
+
+def multiply_letter_pool(LETTER_POOL):
+    weighted_letters =[]
+    for letter, amount in LETTER_POOL.items():
+        grouped_letters = letter * amount
+        for letters in convert_to_list(grouped_letters):
+            weighted_letters.append(letters)
+    return weighted_letters
+
+
+def draw_hand():
+    final_letter_pool = multiply_letter_pool(LETTER_POOL)
+    pool_length = list_length(final_letter_pool)
+    drawn_letters = []
+    purgatory_hand = {}
+    drawn_letters_count = 0
+    while drawn_letters_count != 10:
+        random_index = random.randint(0, pool_length - 1)
+        letter = final_letter_pool[random_index]
+        
+        letter_count = 0
+        if letter in purgatory_hand:
+            letter_count = purgatory_hand[letter]
+        
+        if letter_count < LETTER_POOL[letter]:
+            drawn_letters.append(letter)
+            purgatory_hand[letter] = letter_count + 1
+            drawn_letters_count += 1
+            
+    return drawn_letters
+
+def convert_to_dictionary(list):
+    dictionary = {}
+    for letter in list:
+        if letter in dictionary:
+            dictionary[letter] += 1
+        else:
+            dictionary[letter] = 1
+    return dictionary
+
+def compare_word_to_hand(word_dict, hand_dict):
+    for letter in word_dict:
+        if letter not in hand_dict:
+            return False
+        elif word_dict[letter] > hand_dict[letter]:
+            return False
+    return True
```

Comment:
Looks good to me! The only thing I see is that because we have the same outcome for both the if and elif conditions, we could just condense them into one compound conditional. Other than that though, this looks great. I am always a fan of exiting a function early! You've correctly identified that once you find a bad value, we don't need to worry about the rest of the word!